pandas/io/formats/latex.py

```diff
@@ -256,88 +254,394 @@ def _format_multirow(
+    Parameters
+    ----------
+    formatter : `DataFrameFormatter`
+    column_format : str, default None
+        The columns format as specified in `LaTeX table format
+        <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g 'rcl' for 3 columns
 
-    def _write_tabular_end(self, buf):
+    See Also
+    --------
+    HTMLFormatter
+    """
+
+    def __init__(
+        self,
+        formatter: DataFrameFormatter,
+        longtable: bool = False,
+        column_format: Optional[str] = None,
+        multicolumn: bool = False,
+        multicolumn_format: Optional[str] = None,
+        multirow: bool = False,
+        caption: Optional[str] = None,
+        label: Optional[str] = None,
+        position: Optional[str] = None,
+    ):
+        self.fmt = formatter
+        self.frame = self.fmt.frame
+        self.longtable = longtable
+        self.column_format = column_format
+        self.multicolumn = multicolumn
+        self.multicolumn_format = multicolumn_format
+        self.multirow = multirow
+        self.caption = caption
+        self.label = label
+        self.position = position
+
+    def write_result(self, buf: IO[str]) -> None:
         """
-        Write the end of a tabular environment or nested table/tabular
-        environment.
+        Render a DataFrame to a LaTeX tabular, longtable, or table/tabular
+        environment output.
+        """
+        table_string = self.builder.product
+        buf.write(table_string)
 
-        Parameters
-        ----------
-        buf : string or file handle
-            File path or object. If not specified, the result is returned as
-            a string.
+    @property
+    def builder(self):
+        """Concrete table builder.
 
+        Returns
+        -------
+        TableBuilder
         """
+        kwargs = dict(
+            formatter=self.fmt,
+            column_format=self.column_format,
+            multicolumn=self.multicolumn,
+            multicolumn_format=self.multicolumn_format,
+            multirow=self.multirow,
+            caption=self.caption,
+            label=self.label,
+            position=self.position,
+        )
         if self.longtable:
-            buf.write("\\end{longtable}\n")
+            return LongTableBuilder(**kwargs)
+        if any([self.caption, self.label, self.position]):
+            return RegularTableBuilder(**kwargs)
+        return TabularBuilder(**kwargs)
+
+    @property
+    def column_format(self):
+        """Column format."""
+        return self._column_format
+
+    @column_format.setter
+    def column_format(self, input_column_format):
+        """Setter for column format."""
+        if input_column_format is None:
+            self._column_format = (
+                self._get_index_format() + self._get_column_format_based_on_dtypes()
+            )
+        elif not isinstance(input_column_format, str):  # pragma: no cover
+            raise AssertionError(
```

Comment:
is this hit in the tests?

Reply:
This is the original implementation.
I have just replaced AssertionError with ValueError and added several a test on this.